docs/en/concepts-and-designs/kubesphere-logging.md

```diff
@@ -0,0 +1,46 @@
+
+- Fluent Bit: log collector deployed on each node.
+- Fluent-bit operator: automates tasks for operating Fluent bit instances.
+- Logsidecar injector: injects sidecar into pods for streaming on-disk files to stdout.
+- Elasticsearch cluster: log storage backend.
+- Curator: performs stale log cleanup.
+
+The diagram below presents the overall logging architecture:
+
+![](../../images/kubesphere-logging-overview.png)
+
+The following sections will detail the design and functionality of each component. 
+
+## Log Collection and Forward
+
+There are two types of logs produced by containers:
+
+- stdout and stderr: written to `.log` files in the host directory `/var/log/containers` by default.
+- application logs: files on pod volume, file locations may vary. 
+
+Fluent Bit serves as the node logging agent. It has access to container log directory and works for applications' stdout and stderr. In KubeSphere, we forked and modified the origin Fluent Bit project fluent/fluent-bit. The difference between fluent/fluent-bit and [kubesphere/fluent-bit](https://github.com/kubesphere/fluent-bit) lies that we add a reload interface to support dynamic configuration (see issue [#365](https://github.com/fluent/fluent-bit/issues/365)).
+ 
+As shown in the diagram below, Fluent-bit controller implements the reload interface and is responsible for the life cycle of fluent-bit process.
+
+![](../../images/kubesphere-logging-fluentbit.png) 
+
+The approach to collect on-disk logs is using a streaming sidecar. The sidecar container streams application logs to its own stdout. In this way, we manage log collection in concert. Besides, [Logsidecar Injector](https://github.com/kubesphere/logsidecar-injector) comes to automate sidecar injection. It takes advantages of `MutatingAdmissionWebhook` to control pod creation and modify pod templates.
+
+Though Fluent Bit writes log to Elasticsearch by default in KubeSphere, you may change its output via editing the custom resource defined by the CustomResourceDefinition `fluentbits.logging.kubesphere.io`. The custom resource holds Fluent Bit configuration knowledge. Once changes detected, [Fluent-bit operator](https://github.com/kubesphere/fluentbit-operator) will trigger reloading and the new configuration will be used. The Operator pattern is the encouraged way to implement declarative configuration in Kubernetes. 
+
+## Log Store and Management
+
+Fluent Bit supports a bunch of [output plugins](https://docs.fluentbit.io/manual/output) that you can choose from to configure how logs will be stored or forward. On KubeSphere console, you can easily setup [elasticsearch](https://docs.fluentbit.io/manual/output/elasticsearch), [kafka](https://docs.fluentbit.io/manual/output/kafka) and [fluentd](https://docs.fluentbit.io/manual/output/forward) as outputs.
```

Comment:
"to configure how logs will be stored or forward" => "to configure how logs will be stored or forwarded"
"you can easily setup ...  as outputs." => "you can easily setup ...  as outputs. 0 ~ 3 outputs are allowed to be configured"

Reply:
> 0 ~ 3 outputs are allowed to be configured

The restriction is from the frontend. Maybe we don't need to mention that in backend dev guides